### ISSUE TITLE
Add cache status / cache info CLI inspection command

### ISSUE THREAD

**Issue by vugi99** [vugi99/nanos-lint#14]
### Summary

Add a `cache status` (or `cache info`) inspection command to the CLI to provide clear visibility into installed LuaLS versions, annotations commit SHA, metadata freshness, and disk usage.

---

### Motivation & Background

Currently, the cache directory (`~/.cache/nanos-lint` on Linux/macOS or `AppData\Local\nanos-lint` on Windows) operates as a black box:
- Users have no quick CLI command to see what versions of LuaLS are cached or if they are valid.
- Users cannot check what version or commit of `annotations.lua` is actively cached.
- When troubleshooting unexpected diagnostics or disk usage, users are forced to manually find and inspect hidden system paths or run `clean-cache`.

---

### Proposed Solution

Add the `cache status` command in `src/cli.ts`:

```bash
nanos-lint cache status [options]
# Aliases:
nanos-lint cache info
nanos-lint cache-status
```

#### Options:
- `--json`: Output cache state in JSON format for automated scripting and CI health checks.

#### Example Pretty Output:
```text
nanos-lint Cache Status

Cache Directory:   /home/user/.cache/nanos-lint
Total Disk Usage:  86.4 MB

Lua Language Server (LuaLS):
  - Weekly Check:   2026-W39 (last checked: 2026-09-23)
  - Target Version: 3.19.1
  - Cached Copies:
      * 3.19.1 (status: valid, size: 42.5 MB)

nanos world Annotations:
  - Status:         Valid (1.42 MB)
  - Commit SHA:     8f3a9b2c (docgen-output)
  - Last Checked:   2026-09-23
```

---

### Action Items & Quality Checklist

- [ ] Add `cache status` command in `src/cli.ts`.
- [ ] Implement inspection helpers in `src/paths.ts` or `src/luals/cache.ts` (calculating total directory size, listing cached binaries, inspecting metadata).
- [ ] Support `--json` flag for structured output.
- [ ] Document the command in `README.md` under CLI Reference.
- [ ] Add unit tests in `tests/unit/cli.test.ts`.
- [ ] Verify `npm run lint`, `npm run typecheck`, `npm run build`, and `npm run test:coverage` pass.
- [ ] Update `CHANGELOG.md` under `[Unreleased]` / `Added`.
